ext/pdo_firebird/firebird_driver.c

```diff
@@ -809,28 +806,84 @@ static bool firebird_handle_begin(pdo_dbh_t *dbh) /* {{{ */
 }
 /* }}} */
 
-/* called by PDO to commit a transaction */
-static bool firebird_handle_commit(pdo_dbh_t *dbh) /* {{{ */
+/* called by PDO to start a transaction */
+static bool firebird_handle_manually_begin(pdo_dbh_t *dbh) /* {{{ */
 {
 	pdo_firebird_db_handle *H = (pdo_firebird_db_handle *)dbh->driver_data;
 
-	if (isc_commit_transaction(H->isc_status, &H->tr)) {
-		php_firebird_error(dbh);
+	if (dbh->auto_commit && H->tr && !php_firebird_commit_transaction(dbh, FB_TXN_RELEASE)) {
```

Comment:
```suggestion
php_firebird_commit_transaction(dbh, FB_TXN_RELEASE)
```

I'm confused, ``FB_TXN_RELEASE`` is it's own enum and not a bool, so this is slightly confusing